hedera-node/src/main/java/com/hedera/services/fees/calculation/AutoRenewCalcs.java

```diff
@@ -52,70 +59,141 @@ public class AutoRenewCalcs {
+			this.secondAccountConstantFee = constantFeeFrom(rightPrices);
+			this.firstAccountRbhPrice = leftPrices.getServicedata().getRbh();
+			this.secondAccountRbhPrice = rightPrices.getServicedata().getRbh();
 		}
 	}
 
-	public RenewAssessment maxRenewalAndFeeFor(
-			MerkleAccount expiredAccount,
-			long reqPeriod,
-			Instant at,
-			ExchangeRate rate
+	public void setContractRenewalPriceSeq(
+			final Triple<Map<SubType, FeeData>, Instant, Map<SubType, FeeData>> contractPricesSeq
 	) {
-		if (cryptoAutoRenewPriceSeq == null) {
-			throw new IllegalStateException("No crypto usage prices are set!");
+		this.contractPricesSeq = contractPricesSeq;
+		if (contractPricesSeq.getLeft() == null) {
+			log.warn("No prices known for ContractAutoRenew, will charge zero fees!");
+		} else {
+			final var leftPrices = contractPricesSeq.getLeft().get(SubType.DEFAULT);
+			final var rightPrices = contractPricesSeq.getLeft().get(SubType.DEFAULT);
+			this.firstContractConstantFee = constantFeeFrom(leftPrices);
+			this.secondContractConstantFee = constantFeeFrom(rightPrices);
+			this.firstContractRbhPrice = leftPrices.getServicedata().getRbh();
+			this.firstContractSbhPrice = leftPrices.getServicedata().getSbh();
+			this.secondContractRbhPrice = rightPrices.getServicedata().getRbh();
+			this.secondContractSbhPrice = rightPrices.getServicedata().getSbh();
 		}
+	}
 
-		final long balance = expiredAccount.getBalance();
+	public RenewAssessment assessCryptoRenewal(
+			final MerkleAccount expiredAccountOrContract,
+			final long reqPeriod,
+			final Instant at,
+			final ExchangeRate rate
+	) {
+		final long balance = expiredAccountOrContract.getBalance();
 		if (balance == 0L) {
 			return NO_RENEWAL_POSSIBLE;
 		}
+		final var renewalFees = renewalFees(at, rate, expiredAccountOrContract);
+		return assess(renewalFees, reqPeriod, balance);
+	}
+
+	private RenewalFees renewalFees(
+			final Instant at,
+			final ExchangeRate rate,
+			final MerkleAccount expiredAccountOrContract
+	) {
+		if (expiredAccountOrContract.isSmartContract()) {
+			return contractRenewalPrices(at, rate, expiredAccountOrContract);
+		} else {
+			return accountRenewalPrices(at, rate, expiredAccountOrContract);
+		}
+	}
+
+	private RenewalFees contractRenewalPrices(
+			final Instant at,
+			final ExchangeRate rate,
+			final MerkleAccount contract
+	) {
+		if (contractPricesSeq == null) {
+			throw new IllegalStateException("No contract usage prices are set!");
+		}
 
-		final boolean isBeforeSwitch = at.isBefore(cryptoAutoRenewPriceSeq.getMiddle());
-		final long nominalFixed = isBeforeSwitch ? firstConstantCryptoAutoRenewFee : secondConstantCryptoAutoRenewFee;
-		final long serviceRbhPrice = isBeforeSwitch ? firstServiceRbhPrice : secondServiceRbhPrice;
+		final boolean isBeforeSwitch = at.isBefore(contractPricesSeq.getMiddle());
+		final long fixedPrice = isBeforeSwitch ? firstContractConstantFee : secondContractConstantFee;
```

Comment:
Does this mean if it is before `autoRenewal`, `firstContractConstantFee` is charged , and after `autoRenewal` `secondContractConstantFee` is charged ?

Reply:
Since our [deployed `CurrentAndNextFeeSchedule`](https://hashgraph.github.io/hedera-protobufs/#proto.CurrentAndNextFeeSchedule) always uses the same price for `currentFeeSchedule` and `nextFeeSchedule`, in practice it won't matter.

So this check is just in case these schedules **are** different, and consensus time has passed the [`expiryTime` of the `currentFeeSchedule`](https://hashgraph.github.io/hedera-protobufs/#proto.FeeSchedule).

Reply:
I see .. thank you!